components/drawer/DrawerPanel.tsx

```diff
@@ -72,13 +77,30 @@ const DrawerPanel: React.FC<DrawerPanelProps> = (props) => {
   } = props;
   const drawerContext = useComponentConfig('drawer');
 
+  const closablePlacement = React.useMemo(() => {
```

Comment:
It already has `useClosable` hooks to convert the `closable` to object type. Do not use self check here.

Reply:
You can also HOC the hooks with your version as `useDrawerClosable`